pkg/integrations/v2/app_o11y_receiver/app_o11y_receiver.go

```diff
@@ -0,0 +1,241 @@
+	"github.com/slok/go-http-metrics/middleware"
+	"github.com/slok/go-http-metrics/middleware/std"
+	"go.opentelemetry.io/collector/component"
+)
+
+// IntegrationName is the name of this integration
+var IntegrationName = "app_o11y_receiver"
+
+// Config structs controls the configuration of the app o11y
+// integration
+type Config struct {
+	ReceiverConfig config.AppO11yReceiverConfig `yaml:",inline"`
+	Common         common.MetricsConfig         `yaml:",inline"`
+}
+
+// ApplyDefaults applies runtime-specific defaults to c.
+func (c *Config) ApplyDefaults(globals integrations.Globals) error {
+	c.Common.ApplyDefaults(globals.SubsystemOpts.Metrics.Autoscrape)
+	if id, err := c.Identifier(globals); err == nil {
+		c.Common.InstanceKey = &id
+	}
+	return nil
+}
+
+// Name returns the name of the integration that this config represents
+func (c *Config) Name() string { return IntegrationName }
+
+// Identifier uniquely identifies the app o11y integration
+func (c *Config) Identifier(globals integrations.Globals) (string, error) {
+	if c.Common.InstanceKey != nil {
+		return *c.Common.InstanceKey, nil
+	}
+	return globals.AgentIdentifier, nil
+}
+
+type appo11yIntegration struct {
+	integrationName, instanceID string
+	globals                     integrations.Globals
+	logger                      log.Logger
+	conf                        config.AppO11yReceiverConfig
+	common                      common.MetricsConfig
+	handler                     handler.AppO11yHandler
+	exporters                   []exporters.AppO11yReceiverExporter
+	reg                         *prometheus.Registry
+}
+
+// Static typecheck tests
+var (
+	_ integrations.Integration        = (*appo11yIntegration)(nil)
+	_ integrations.HTTPIntegration    = (*appo11yIntegration)(nil)
+	_ integrations.MetricsIntegration = (*appo11yIntegration)(nil)
+)
+
+// NewIntegration converts this config into an instance of an integration
+func (c *Config) NewIntegration(l log.Logger, globals integrations.Globals) (integrations.Integration, error) {
+	id, err := c.Identifier(globals)
+	if err != nil {
+		return nil, err
+	}
+
+	reg := prometheus.NewRegistry()
+
+	if err != nil {
+		return nil, err
+	}
+	sourcemapLogger := log.With(l, "subcomponent", "sourcemaps")
+	sourcemapStore := sourcemaps.NewSourceMapStore(sourcemapLogger, c.ReceiverConfig.SourceMaps, reg, nil, nil)
+
+	receiverMetricsExporter := exporters.NewReceiverMetricsExporter(exporters.ReceiverMetricsExporterConfig{
+		Reg: reg,
+	})
+
+	var exp = []exporters.AppO11yReceiverExporter{
+		receiverMetricsExporter,
+	}
+
+	if len(c.ReceiverConfig.LogsInstance) > 0 {
+		logsInstance := globals.Logs.Instance(c.ReceiverConfig.LogsInstance)
+		if logsInstance == nil {
+			return nil, fmt.Errorf("logs instance \"%s\" not found", c.ReceiverConfig.LogsInstance)
+		}
+		lokiExporter := exporters.NewLogsExporter(
+			l,
+			exporters.LogsExporterConfig{
+				LogsInstance:     logsInstance,
```

Comment:
You can't hold a pointer to a logs instance throughout the lifetime of an integration; the implementation of the logs subsystem will create new pointers if the logs instance changes. 

That means you have to do a lookup by name any time the integration needs to access the instance. The same should be done for metrics and tracing (when added).